src/mdns/mdns_mdnssd.cpp

```diff
@@ -361,16 +428,20 @@ void PublisherMDnsSd::HandleServiceRegisterResult(DNSServiceRef         aService
 {
     OTBR_UNUSED_VARIABLE(aDomain);
 
-    otbrError       error = DNSErrorToOtbrError(aError);
-    std::string     originalInstanceName;
-    ServiceIterator service = FindPublishedService(aServiceRef);
+    std::string            type = aType;
+    std::string            originalInstanceName;
+    ServiceRegistrationPtr serviceReg = FindServiceRegistration(aServiceRef);
+
+    VerifyOrExit(serviceReg != nullptr);
 
-    VerifyOrExit(service != mServices.end());
+    if (type.back() == '.')
+    {
+        type.pop_back();
+    }
 
-    // mDNSResponder could auto-rename the service instance name when name conflict
-    // is detected. In this case, `aName` may not match `service->mName` and we
-    // should use the original `service->mName` to find associated SRP service.
-    originalInstanceName = service->mName;
+    // mDNSResponder could auto-rename the service instance
+    // name when name conflict is detected.
```

Comment:
Is it still true? I saw you've turned on `kDNSServiceFlagsNoAutoRename`.

Reply:
Good catch! I cleaned up this logic.